stdlib/string.mc

```diff
@@ -29,6 +31,63 @@ let int2string = fix (lam int2string. lam n.
     concat (int2string (divi n 10)) d
 )
 
+// Returns an option with the index of the first occurrence of c in s. Returns
+// None if c was not found in s.
+let strindex = lam c. lam s.
+  let strindex_abs = fix (lam strindex. lam i. lam c. lam s.
+    if eqi (length s) 0
+    then None
+    else if eqchar c (head s)
+         then Some(i)
+         else strindex (addi i 1) c (tail s)
+  ) in
+  strindex_abs 0 c s
+
+// Returns an option with the index of the last occurrence of c in s. Returns
+// None if c was not found in s.
+let strrindex = lam c. lam s.
+  let strrindex_abs = fix (lam strrindex. lam i. lam acc. lam c. lam s.
+    if eqi (length s) 0 then
+      if eqi acc (negi 1)
+      then None
+      else Some(acc)
+    else
+      if eqchar c (head s)
+      then strrindex (addi i 1) i   c (tail s)
+      else strrindex (addi i 1) acc c (tail s)
+  ) in
+  strrindex_abs 0 (negi 1) c s
```

Comment:
May I suggest a name that is more than one letter different from `strindex`? Maybe call them `strIndex` and `strLastIndex`?

Reply:
(or non-camelCase, if you prefer. We're going to have to go over the whole standard library at some later point to make naming convention the same all over)

Reply:
Done. I usually prefer a non-camelCase naming scheme, but it is better to have a consistent naming scheme (I saw that you use camelCase for `zipWith` etc.)

Reply:
Awesome! Merging now...